internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -102,15 +107,22 @@ func (provisioner *KopsProvisioner) CreateCluster(cluster *model.Cluster, awsCli
 		clusterResources.WorkerSecurityGroupIDs,
 		allowSSHCIDRS,
 	)
-	if err != nil {
+	// Only release VPC resources, if create cluster request was for the primary cluster.
+	if err != nil && kopsMetadata.ChangeRequest.VPC == "" {
 		releaseErr := awsClient.ReleaseVpc(cluster.ID, logger)
 		if releaseErr != nil {
 			logger.WithError(releaseErr).Error("Unable to release VPC")
 		}
 
 		return errors.Wrap(err, "unable to create kops cluster")
 	}
-
+	// Tag Public subnets & respective VPC for the secondary cluster.
+	if err == nil && kopsMetadata.ChangeRequest.VPC != "" {
```

Comment:
Same here I think.

Reply:
Same err check not required. 
modify the condition and did a quick cluster/ delete test. looks fine to me.